src/components/FormFieldRecommendedLengthInput/FormFieldRecommendedLengthInput.vue

```diff
@@ -0,0 +1,171 @@
+<template>
+  <FormGroup>
+    <InputLabel v-if="Boolean(label)" :for="name">{{ label }}</InputLabel>
+
+    <BaseInput
+      v-if="type !== 'textarea'"
+      :id="name"
+      :name="name"
+      :type="type"
+      :value="value"
+      v-bind="$attrs"
+      v-on="$listeners"
+    />
+
+    <BaseTextArea
+      v-else
+      :id="name"
+      :name="name"
+      :type="type"
+      :value="value"
+      v-bind="$attrs"
+      v-on="$listeners"
+    />
+
+    <Progress :percent="percent" :color="color" />
+
+    <div class="count">
+      <span class="count-zero">0</span>
+
+      <span
+        v-if="minLength"
+        class="count-lower-limit"
+        :style="lowerLimitStyles"
+      >
+        {{ minLength }}
+      </span>
+
+      <span v-if="maxLength" class="count-upper-limit">{{ maxLength }}</span>
+    </div>
+
+    <FormFieldError v-if="Boolean(error)">{{ error }}</FormFieldError>
+  </FormGroup>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent } from '@vue/composition-api';
+
+import FormGroup from '../FormGroup.vue';
+import InputLabel from '../InputLabel/index.vue';
+import BaseInput from '../BaseInput';
+import BaseTextArea from '../BaseTextArea';
+import FormFieldError from '../FormFieldError/index.vue';
+import Progress from './components/Progress';
+
+interface Props {
+  name: string;
+  label: string;
+  value: string;
+  type: string;
+  error: string;
+  minLength: number;
+  maxLength: number;
+}
+
+export default defineComponent<Props>({
+  name: 'FormFieldRecommendedLengthInput',
+  components: {
+    FormGroup,
+    InputLabel,
+    BaseInput,
+    BaseTextArea,
+    FormFieldError,
+    Progress,
+  },
+  inheritAttrs: false,
+  props: {
+    name: {
+      type: String,
+      required: true,
+    },
+    label: String,
+    value: {
+      type: String,
+      default: '',
+    },
+    type: {
+      type: String,
+      default: 'text',
+      validator(value: string) {
+        return [
+          'text',
+          'number',
+          'email',
+          'password',
+          'date',
+          'time',
+          'textarea',
+        ].includes(value);
+      },
+    },
+    error: String,
+    minLength: Number,
+    maxLength: Number,
+  },
+  setup(props: Props) {
+    const isLowerThanMinLength = computed<boolean>(() => {
+      return props.value.length < props.minLength;
+    });
+
+    const isHigherThanMaxLength = computed<boolean>(() => {
+      return props.value.length > props.maxLength;
+    });
+
+    const percent = computed<number>(() => {
+      const A1: number = props.value.length;
+      const A2: number = props.maxLength ? props.maxLength : props.minLength;
+      return (A1 / A2) * 100;
+    });
+
+    const color = computed<string>(() => {
```

Comment:
add comment here
`// TODO Refactor it. Maybe move this logic to Progress component`